integrations/meta_llama.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: integration
+name: Meta Llama API
+description: Use Llama Models with Haystack
+authors:
+    - name: Young Han
+      socials:
+        github: https://github.com/seyeong-han
+        twitter: deepset_ai
+        linkedin: https://www.linkedin.com/company/deepset-ai/
+pypi: https://pypi.org/project/meta-llama-haystack/
+repo: https://github.com/deepset-ai/haystack-core-integrations/tree/main/integrations/meta_llama
+type: Model Provider
+report_issue: https://github.com/deepset-ai/haystack-core-integrations/issues
+logo: /logos/meta_llama.png
+version: Haystack 2.0
+toc: true
+---
+
+### **Table of Contents**
+
+- [Overview](#overview)
+- [Installation](#installation)
+- [Usage](#usage)
+
+## Overview
+
+This integration supports Meta Llama models provided through Meta’s own inferencing infrastructure. To get the `LLAMA_API_KEY`, check out [the Llama API website](https://llama.developer.meta.com?utm_source=partner-haystack&utm_medium=website).
+
+You can use Llama models with `MetaLlamaChatGenerator`.
+
+Currently, available models are:
+
+| Model ID | Input context length | Output context length | Input Modalities | Output Modalities |
+| --- | --- | --- | --- | --- |
+| `Llama-4-Scout-17B-16E-Instruct-FP8` | 128k | 4028 | Text, Image | Text |
+| `Llama-4-Maverick-17B-128E-Instruct-FP8` | 128k | 4028 | Text, Image | Text |
+| `Llama-3.3-70B-Instruct` | 128k | 4028 | Text | Text |
+| `Llama-3.3-8B-Instruct` | 128k | 4028 | Text | Text |
+
+## Installation
+
+```bash
+pip install meta-llama-haystack
+```
+
+## Usage
+
+Based on your use case, you can choose between `MetaLlamaChatGenerator`.
+Before using, make sure to set the `LLAMA_API_KEY` environment variable.
+
+### Using `MetaLlamaChatGenerator`
+
+This example showcases how to build a complete RAG system that can answer questions based on the information in your document store using Meta's Llama models.
+
+```python
+# To run this example, you will need to set a `LLAMA_API_KEY` environment variable.
+
+# Copyright (c) Meta Platforms, Inc. and affiliates
```

Comment:
```suggestion
```
This code example is a minor modification of an example from the haystack-integrations repository, with only the line `from haystack_integrations.components.generators.meta_llama import MetaLlamaChatGenerator` added and the new class MetaLlamaChatGenerator used instead of other chat generators. Adding a copyright claim like “Copyright (c) Meta Platforms, Inc. and affiliates” here can be misleading, as it implies original authorship of the overall code example.